### PR TITLE
Lobby required displayname

### PR DESCRIPTION
**Pull request by damencho**

### CODE REVIEW

File: resources/prosody-plugins/mod_muc_lobby_rooms.lua [hunk 1]
```diff
@@ -42,7 +45,14 @@ if lobby_muc_component_config == nil then
     return ;
 end
 
-local whitelist = module:get_option_set("muc_lobby_whitelist", {});
+local whitelist;
+local check_display_name_required;
+local function load_config()
+    whitelist= module:get_option_set("muc_lobby_whitelist", {});
```
Comment (bgrozev) [inline resources/prosody-plugins/mod_muc_lobby_rooms.lua:51]: nit: Whitespace

Comment (damencho) [reply resources/prosody-plugins/mod_muc_lobby_rooms.lua]: Fixed

File: resources/prosody-plugins/mod_auth_token.lua [hunk 2]
```diff
@@ -31,9 +31,13 @@ function init_session(event)
         -- (see get_username_from_token)
         session.previd = query and params.previd or nil;
 
-        -- The room name and optional prefix from the bosh query
-        session.jitsi_bosh_query_room = params.room;
-        session.jitsi_bosh_query_prefix = params.prefix or "";
+        -- The room name and optional prefix from the web query
```
Comment (bgrozev) [inline resources/prosody-plugins/mod_auth_token.lua:34]: Can you add a note that this field is not verified by the token?

File: resources/prosody-plugins/mod_muc_lobby_rooms.lua [hunk 3]
```diff
@@ -86,6 +96,11 @@ function filter_stanza(stanza)
             return nil;
         end
 
+        -- allow disco info from the lobby component
+        if stanza.name == 'iq' and stanza:get_child('query', DISCO_INFO_NS) then
```
Comment (bgrozev) [inline resources/prosody-plugins/mod_muc_lobby_rooms.lua:100]: Use `else if`?

**Comment by damencho** [comment]
Lobby tests are work in progress.

**Pull request merged**